src/main/java/org/frc5687/powerup/robot/Robot.java

```diff
@@ -149,10 +149,10 @@ public void testPeriodic() {
 
     public void updateDashboard() {
         pdp.updateDashboard();
-        intake.updateDashboard();
-        driveTrain.updateDashboard();
-        carriage.updateDashboard();
-        _arm.updateDashboard();
+        intake.periodic();
```

Comment:
iirc you should not have to call this manually. 

If you take a look at the [javadoc for the wpilib](http://first.wpi.edu/FRC/roborio/release/docs/java/) for Subsystem you will see that the method `periodic` has the description of:

> When the run method of the scheduler is called this method will be called.

If you are unsure of what the run method is, then you can look at https://github.com/frc5687/2018-robot/blob/master/src/main/java/org/frc5687/powerup/robot/Robot.java#L144, where we run `Scheduler.getInstance().run()`. If you follow this, you will find the `run()` method in `Scheduler`.

```
/**
   * Runs a single iteration of the loop. This method should be called often in order to have a
   * functioning {@link Command} system. The loop has five stages:
   *
   * <ol> <li>Poll the Buttons</li> <li>Execute/Remove the Commands</li> <li>Send values to
   * SmartDashboard</li> <li>Add Commands</li> <li>Add Defaults</li> </ol>
   */
  public void run() {

    m_runningCommandsChanged = false;

    if (m_disabled) {
      return;
    } // Don't run when m_disabled

    // Get button input (going backwards preserves button priority)
    if (m_buttons != null) {
      for (int i = m_buttons.size() - 1; i >= 0; i--) {
        m_buttons.elementAt(i).execute();
      }
    }

    // Call every subsystem's periodic method
    Enumeration subsystems = m_subsystems.getElements();
    while (subsystems.hasMoreElements()) {
      ((Subsystem) subsystems.nextElement()).periodic();
    }

    // Loop through the commands
    LinkedListElement element = m_firstCommand;
    while (element != null) {
      Command command = element.getData();
      element = element.getNext();
      if (!command.run()) {
        remove(command);
        m_runningCommandsChanged = true;
      }
    }

    // Add the new things
    for (int i = 0; i < m_additions.size(); i++) {
      _add(m_additions.elementAt(i));
    }
    m_additions.removeAllElements();

    // Add in the defaults
    Enumeration locks = m_subsystems.getElements();
    while (locks.hasMoreElements()) {
      Subsystem lock = (Subsystem) locks.nextElement();
      if (lock.getCurrentCommand() == null) {
        _add(lock.getDefaultCommand());
      }
      lock.confirmCommand();
    }
  }
```

where in these lines

```
// Call every subsystem's periodic method
    Enumeration subsystems = m_subsystems.getElements();
    while (subsystems.hasMoreElements()) {
      ((Subsystem) subsystems.nextElement()).periodic();
    }
```

you can see that `periodic()` is already called by the scheduler's `run()` method, so you should not need to call `periodic()` manually.